### ISSUE TITLE
Interoperability Chrome, Firefox, Opera

### ISSUE THREAD

**Issue by mrbatista** [ging/licode#299]
I have tested this browser on the demo [page](http://chotis2.dit.upm.es/), the scenario is the following: three users each with a different browser. The result is (on a MacBook Pro Retina Yosemite 10.10):

1. Room NO p2p: the video call works perfectly; all subscribers receive others' audio/video streams;
2. Room p2p: only Firefox receives all two subscribers, Chrome and Opera do not see Firefox subscribers. No events is fired.

Has anyone encountered this problem with a p2p room?
Version browser:
Chrome: 43.0.2357.65 (64-bit)
Firefox: 38.0.1
Opera: 29.0.1795.60

**Comment by mrbatista** [comment]
The problem with p2p room is also present in my development environment with master branch

**Comment by mrbatista** [comment]
I've investigated and the interoperability Chrome (or Opera) <-> Firefox is broken.
Error is: "setting Remote Description Failed to parse SessionDescription. 
a=rtcp-fb:120 nack pli Invalid SDP line."
The problem seems to be the function [changeOrder](https://github.com/ging/licode/blob/master/erizo_controller/erizoClient/src/webrtc-stacks/FirefoxStack.js#L110) in FirefoxStack.js
If remove this function i have black stream in Chrome and video stream in Firefox.

**Comment by mrbatista** [comment]
@lodoyun any suggestions to resolve this issue?

**Comment by zevarito** [comment]
@mrbatista newbie question: how you determine which is P2P and which is not?

**Comment by mrbatista** [comment]
@zevarito if room is p2p Licode not use MCU. All token created with p2p room not use MCU. This is the logic implemented in Licode.
If you see SDP there is a line that identify a SDP forma MCU with "LicodeMCU"

**Comment by zevarito** [comment]
@mrbatista 

For sure I have not clear the functionality of MCU.

* It is possible for me to select which one to use?
* Is Licode the one who decides if MCU is used or not?

THanks!

**Comment by nachoaguadoc** [comment]
Hi @zevarito 

If you want to create a P2P room you have to include this option as a parameter when you call Nuve in the application script:
```
N.API.createRoom( "MyP2PRoom", function(room) {
      console.log("P2P room created with ID:", room._id);
}, errorCallback, {p2p: true});
```

**Comment by mrbatista** [comment]
@lodoyun I've update to master and I did some testing:
Firefox Desktop <-> Chrome Desktop **broken**; (Mac Yosemite OS X 10.10.3)
Chrome Desktop <-> Firefox Mobile **broken**; (Mac Yosemite OS X 10.10.3)
Firefox Desktop <-> Chrome Mobile **works**; (Mac Yosemite OS X 10.10.3, Android 5.0)
Firefox Desktop <-> Firefox Mobile **works**; (Mac Yosemite OS X 10.10.3, Android 5.0)
Chrome Desktop <-> Chrome Mobile  **works**; (Mac Yosemite OS X 10.10.3, Android 5.0)

Later I'll test with Windows OS. I have private branch with some cleanup, but problem persist.

@zevarito 
1) It's possible to select MCU or p2p. See example of @nachoaguadoc 
2) YOU decide if create a p2p room or not. Room not p2p for example permit to record streams...

**Comment by zevarito** [comment]
Thanks @nachoaguadoc @mrbatista!

@mrbatista, I guessed that p2p rooms were not able to be recorded, but at the same time I thought MCU acts as simple stream consumer to make a recording. Is that right? MCU recording capability are the same that if any of the peers want to record the stream from client side?

**Comment by zevarito** [comment]
@mrbatista I could confirm FF Desk -> CH Desk doesn't work on master, I've updated our servers two days ago. Before that we didn't had FF compatibility so I could not tell you if that was working a few commits behind master.

**Comment by lodoyun** [comment]
It took a while but it's fixed for me now in commit 9f426b64b0188e6cd08476b443032ef4303ff6ba .
Firefox didn't like it when we added remote Ice candidates before it generated it's first local candidate. There was a race condition.
